src/sound.c

```diff
@@ -385,30 +385,23 @@ f_sound_playfile(typval_T *argvars, typval_T *rettv)
     if (in_vim9script() && check_for_string_arg(argvars, 0) == FAIL)
 	return;
 
-    esc = vim_strsave_shellescape(tv_get_string(&argvars[0]), FALSE, FALSE);
+    filepath = tv_get_string(&argvars[0]);
 
-    len = STRLEN(esc) + 5 + 18 + 1;
+    len = STRLEN(filepath) + 5 + 18 + 2 + 1;
     p = alloc(len);
     if (p == NULL)
     {
-	free(esc);
 	return;
     }
-    vim_snprintf((char *)p, len, "open %s alias sound%06ld", esc, newid);
-    free(esc);
+    vim_snprintf((char *)p, len, "open \"%s\" alias sound%06ld", filepath, newid);
 
-    wp = enc_to_utf16((char_u *)p, NULL);
+    err = mciSendStringA(p, NULL, 0, sound_window());
     free(p);
-    if (wp == NULL)
-	return;
-
-    err = mciSendStringW(wp, NULL, 0, sound_window());
-    free(wp);
```

Comment:
This part should be kept. Otherwise, Unicode filenames cannot be handled.

Reply:
Right. I thought that now that windows and linux use utf-8 as multibyte character set that was no longer required. But after some testing is clear that I was wrong.